catalogs/catalog-lakehouse-hudi/src/main/java/org/apache/gravitino/catalog/lakehouse/hudi/backend/hms/HudiHMSBackendOps.java

```diff
@@ -73,12 +125,51 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
 
   @Override
   public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaException {
-    throw new UnsupportedOperationException("Not implemented yet");
+    NameIdentifier schemaIdent = NameIdentifier.of(namespace.levels());
+    if (!schemaExists(schemaIdent)) {
+      throw new NoSuchSchemaException("Schema (database) does not exist %s", namespace);
+    }
+
+    try {
+      return clientPool.run(
+          c ->
+              c.getTables(schemaIdent.name(), "*").stream()
+                  .map(table -> NameIdentifier.of(namespace, table))
+                  .toArray(NameIdentifier[]::new));
```

Comment:
Will here list all the tables, not just Hudi table, should we filter out non-hudi table?

Reply:
fixed